hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/validation/TokenValidator.java

```diff
@@ -66,26 +67,32 @@ public TokenValidator(Validator validator) throws MalformedURLException {
 
 	@Override
 	public boolean validate(Exchange exchange) throws Exception {
-		logger.info("TokenValidator validation started");
-
 		String methodName = LoggingUtil.getMethodName();
+
+		logger.debug("{} - TransactionId: {}, TokenValidator validation started", methodName, exchange.getExchangeId());
 		
 		// If more validataion is required for exchange message, we should create a new bean
 		String authorizationKey = (String) exchange.getIn().getHeader(AUTHORIZATION);
-		if(StringUtils.isBlank(authorizationKey)) {
+		if (StringUtils.isBlank(authorizationKey)) {
 			logger.info("{} - TransactionId: {}, No authorization key passed in request header.", methodName, exchange.getExchangeId());
-			throw new CustomHNSException(CustomError_Msg_InvalidAuthKey.getErrorMessage());
+			throw new CustomHNSException(CustomError_Msg_MissingAuthKey);
 		}
-		AccessToken accessToken = AccessToken.parse(authorizationKey);
-		logger.info("{} - TransactionId: {}, Access token: {}", methodName,exchange.getExchangeId(),accessToken);
+		try {
+			AccessToken accessToken = AccessToken.parse(authorizationKey);
+			logger.info("{} - TransactionId: {}, Access token: {}", methodName, exchange.getExchangeId(), accessToken);
 
-		// Process the token
-		JWTClaimsSet claimsSet = jwtProcessor.process(accessToken.toString(), null);
+			// Process the token
+			JWTClaimsSet claimsSet = jwtProcessor.process(accessToken.toString(), null);
+
+			// Print out the token claims set
+			logger.info("{} - TransactionId: {}, TOKEN PAYLOAD: {}", methodName, exchange.getExchangeId(), claimsSet.toJSONObject());
+		
+			logger.debug("{} - TransactionId: {}, TokenValidator validation completed", methodName, exchange.getExchangeId());			
+		} catch (ParseException | java.text.ParseException | BadJOSEException | JOSEException e) {
+			logger.error("{} - TransactionId: {}, Error: {}", methodName, exchange.getExchangeId(), e.getMessage());
+			throw new CustomHNSException(ErrorMessage.CustomError_Msg_InvalidAuthKey);
```

Comment:
Added exception handling for better control over which exception is generated. Previous version just allowed exceptions to bubble up.